src/main/java/uk/gov/companieshouse/acsp/service/AcspService.java

```diff
@@ -209,4 +209,15 @@ public ResponseEntity<Object> deleteAcspApplication(String id) {
             return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
         }
     }
+
+    public ResponseEntity<Object> deleteAcspApplicationAndTransaction(String id, String transactionId) {
+        try {
+            acspRepository.deleteById(id);
+            transactionService.deleteTransaction(transactionId);
+            return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+        } catch (Exception e) {
+            LOGGER.error("Error deleting document with id " + id + "and transaction ID: " + transactionId, e);
```

Comment:
Might need to add whitespace between the id + "and transaction ID: " on this log message.

Something like this?
`"Error deleting document with id: " + id + " and transaction ID: " + transactionId, e`